blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/astyanax/AstyanaxReader.java

```diff
@@ -243,64 +205,51 @@ public String getLastStringValue(Locator locator) {
         return points;
     }
 
-    public static String getType(Locator locator) {
-        String type = null;
-        try {
-            type = metaCache.get(locator, MetricMetadata.TYPE.name().toLowerCase(), String.class);
-        } catch (CacheException ex) {
-            log.warn("Cache exception reading type from MetadataCache. ", ex);
+    /**
+     * Get data points for a particular {@link Locator}, {@link Range}, and
+     * {@link Granularity}.
+     *
+     * This method is eventually called from Rollup process.
+     *
+     * @param locator
+     * @param range
+     * @param gran
+     * @return a {@link MetricData} containing the data points requested
+     */
+    public MetricData getDatapointsForRange(Locator locator, Range range, Granularity gran) {
+        RollupType rollupType = null;
+        String rollupTypeStr = metaCache.safeGet(locator, rollupTypeCacheKey);
+        if ( rollupTypeStr != null ) {
+            rollupType = RollupType.fromString(rollupTypeStr);
         }
-        if (type == null) {
-            type = Util.UNKNOWN;
+        if ( rollupType == null ) {
+            rollupType = RollupType.BF_BASIC;
         }
-        return type;
-    }
-
-    public MetricData getDatapointsForRange(Locator locator, Range range, Granularity gran) {
-        try {
-            Object type = metaCache.get(locator, dataTypeCacheKey);
-            RollupType rollupType = RollupType.fromString(metaCache.get(locator, rollupTypeCacheKey));
-
-            if (rollupType == null) {
-                rollupType = RollupType.BF_BASIC;
-            }
-            if (type == null) {
-                return getNumericOrStringRollupDataForRange(locator, range, gran, rollupType);
-            }
-
-            DataType metricType = new DataType((String) type);
-            if (!DataType.isKnownMetricType(metricType)) {
-                return getNumericOrStringRollupDataForRange(locator, range, gran, rollupType);
-            }
-            if (metricType.equals(DataType.STRING)) {
-                gran = Granularity.FULL;
-                return getStringMetricDataForRange(locator, range, gran);
-            } else if (metricType.equals(DataType.BOOLEAN)) {
-                gran = Granularity.FULL;
-                return getBooleanMetricDataForRange(locator, range, gran);
-            } else {
-                return getNumericMetricDataForRange(locator, range, gran, rollupType, metricType);
-            }
 
-        } catch (CacheException e) {
-            log.warn("Caught exception trying to find metric type from meta cache for locator " + locator.toString(), e);
-            return getNumericOrStringRollupDataForRange(locator, range, gran, RollupType.BF_BASIC);
-        }
+        return getNumericMetricDataForRange(locator, range, gran, rollupType);
```

Comment:
nm.  I misread the diff. I thought ``getDatapointsForRange()``` only called ```getNumericMetricDataForRange()```